rclpy/src/rclpy/_rclpy.c

```diff
@@ -546,7 +546,9 @@ _expand_topic_name_with_exceptions(const char * topic, const char * node, const
   if (rcutils_ret != RCUTILS_RET_OK) {
     PyErr_Format(PyExc_RuntimeError, "%s", rcutils_get_error_string_safe());
     rcutils_reset_error();
-    allocator.deallocate(expanded_topic, allocator.state);
+    if (expanded_topic) {
```

Comment:
Using an if on `free` is not necessary/recommended against. I would suggest that we hold the policy that our `deallocate` functions should also have the same behavior as free. And thus not add custom logic outside the deallocator to check before.

Two related threads: 
- https://stackoverflow.com/questions/1938735/does-freeptr-where-ptr-is-null-corrupt-memory
- https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c
Which link to [ISO-IEC 9899](http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf) Section 7.20.3 is relevant.